RDMA: -Werror=pointer-to-int-cast on 32 bit

Building 0.2.1 (with #248) on `i586` and `armv7l` returns `pointer-to-int-cast` warnings-to errors (`-Werror`)  upon building with RDMA:

````
[   14s] [ 63%] Building C object CMakeFiles/valkey_unittest.dir/src/rdma.c.o
[   14s] /usr/bin/cc  -I/home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/include/valkey -I/home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src -fomit-frame-pointer -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -O2 -g -DNDEBUG -std=c99 -Werror -Wall -Wextra -pedantic -Wstrict-prototypes -Wwrite-strings -Wno-missing-field-initializers -MD -MT CMakeFiles/valkey_unittest.dir/src/rdma.c.o -MF CMakeFiles/valkey_unittest.dir/src/rdma.c.o.d -o CMakeFiles/valkey_unittest.dir/src/rdma.c.o -c /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src/rdma.c
[   14s] /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src/rdma.c: In function ‘rdmaPostRecv’:
[   14s] /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src/rdma.c:164:21: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
[   14s]   164 |     recv_wr.wr_id = (uint64_t)cmd;
[   14s]       |                     ^
[   14s] /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src/rdma.c: In function ‘rdmaSendCommand’:
[   14s] /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src/rdma.c:305:21: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
[   14s]   305 |     send_wr.wr_id = (uint64_t)_cmd;
[   14s]       |                     ^
[   14s] In file included from /usr/include/sys/types.h:176,
[   14s]                  from /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/include/valkey/valkey.h:42,
[   14s]                  from /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/include/valkey/async.h:34,
[   14s]                  from /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src/rdma.c:36:
[   14s] /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src/rdma.c: In function ‘connRdmaRegisterRx’:
[   14s] /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src/rdma.c:323:31: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
[   14s]   323 |     cmd.memory.addr = htobe64((uint64_t)ctx->recv_buf);
[   14s]       |                               ^
[   14s] /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src/rdma.c: In function ‘connRdmaHandleRecv’:
[   14s] /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/src/rdma.c:341:24: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
[   14s]   341 |         ctx->tx_addr = (char *)be64toh(cmd->memory.addr);
[   14s]       |                        ^
[   14s] cc1: all warnings being treated as errors
[   14s] make[2]: *** [CMakeFiles/valkey_unittest.dir/build.make:292: CMakeFiles/valkey_unittest.dir/src/rdma.c.o] Error **1**
````


````
[   13s] + /usr/bin/cmake /home/abuild/rpmbuild/BUILD/libvalkey-0.2.1-build/libvalkey-0.2.1/. '-GUnix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_INSTALL_BINDIR:PATH=bin -DCMAKE_INSTALL_SBINDIR:PATH=sbin -DCMAKE_INSTALL_LIBEXECDIR:PATH=libexec -DCMAKE_INSTALL_SYSCONFDIR:PATH=etc -DCMAKE_INSTALL_SHAREDSTATEDIR:PATH=/var/lib -DCMAKE_INSTALL_LOCALSTATEDIR:PATH=var -DCMAKE_INSTALL_RUNSTATEDIR:PATH=run -DCMAKE_INSTALL_LIBDIR:PATH=lib -DCMAKE_INSTALL_INCLUDEDIR:PATH=include -DCMAKE_INSTALL_DATAROOTDIR:PATH=share -DCMAKE_INSTALL_DOCDIR:PATH=share/doc/packages/libvalkey -DCMAKE_INSTALL_MANDIR:PATH=share/man -DCMAKE_INSTALL_INFODIR:PATH=share/info -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DCMAKE_BUILD_TYPE=RelWithDebInfo '-DCMAKE_C_FLAGS=-fomit-frame-pointer -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g' '-DCMAKE_CXX_FLAGS=-fomit-frame-pointer -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g' '-DCMAKE_Fortran_FLAGS=-fomit-frame-pointer -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g' '-DCMAKE_EXE_LINKER_FLAGS=-flto=auto -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' '-DCMAKE_MODULE_LINKER_FLAGS=-flto=auto -Wl,--as-needed' '-DCMAKE_SHARED_LINKER_FLAGS=-flto=auto -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=ON -DBUILD_STATIC_LIBS:BOOL=OFF -DCMAKE_COLOR_MAKEFILE:BOOL=OFF -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_MODULES_INSTALL_DIR=/usr/lib/cmake/libvalkey -DCMAKE_POLICY_VERSION_MINIMUM=3.5 -DENABLE_TLS=1 -DENABLE_RDMA=1
````


This is on openSUSE Tumbleweed, gcc15. Logs currently [here](https://build.opensuse.org/package/show/home:AndreasStieger:branches:server:database/libvalkey)

--- a/src/rdma.c
+++ b/src/rdma.c
@@ -47,6 +47,7 @@
 #include <netdb.h>
 #include <poll.h>
 #include <rdma/rdma_cma.h>
+#include <stdint.h>
 #include <sys/socket.h>
 #include <sys/types.h>
 #include <unistd.h>
@@ -129,6 +130,9 @@ typedef struct RdmaContext {
     struct ibv_mr *cmd_mr;
 } RdmaContext;
 
+/* Apparently CHERI uintptr_t can be 128 bits */
+vk_static_assert(sizeof(uintptr_t) <= sizeof(uint64_t));
+
 static int valkeyRdmaCM(valkeyContext *c, long timeout);
 
 static int valkeyRdmaSetFdBlocking(valkeyContext *c, int fd, int blocking) {
@@ -161,7 +165,7 @@ static int rdmaPostRecv(RdmaContext *ctx, struct rdma_cm_id *cm_id, valkeyRdmaCm
     sge.length = length;
     sge.lkey = ctx->cmd_mr->lkey;
 
-    recv_wr.wr_id = (uint64_t)cmd;
+    recv_wr.wr_id = (uint64_t)(uintptr_t)cmd;
     recv_wr.sg_list = &sge;
     recv_wr.num_sge = 1;
     recv_wr.next = NULL;
@@ -302,7 +306,7 @@ static int rdmaSendCommand(valkeyContext *c, struct rdma_cm_id *cm_id, valkeyRdm
 
     send_wr.sg_list = &sge;
     send_wr.num_sge = 1;
-    send_wr.wr_id = (uint64_t)_cmd;
+    send_wr.wr_id = (uint64_t)(uintptr_t)_cmd;
     send_wr.opcode = IBV_WR_SEND;
     send_wr.send_flags = IBV_SEND_SIGNALED;
     send_wr.next = NULL;
@@ -320,7 +324,7 @@ static int connRdmaRegisterRx(valkeyContext *c, struct rdma_cm_id *cm_id) {
     valkeyRdmaCmd cmd = {0};
 
     cmd.memory.opcode = htons(RegisterXferMemory);
-    cmd.memory.addr = htobe64((uint64_t)ctx->recv_buf);
+    cmd.memory.addr = htobe64((uint64_t)(uintptr_t)ctx->recv_buf);
     cmd.memory.length = htonl(ctx->recv_length);
     cmd.memory.key = htonl(ctx->recv_mr->rkey);
 
@@ -338,7 +342,7 @@ static int connRdmaHandleRecv(valkeyContext *c, RdmaContext *ctx, struct rdma_cm
 
     switch (ntohs(cmd->keepalive.opcode)) {
     case RegisterXferMemory:
-        ctx->tx_addr = (char *)be64toh(cmd->memory.addr);
+        ctx->tx_addr = (char *)(uintptr_t)be64toh(cmd->memory.addr);
         ctx->tx_length = ntohl(cmd->memory.length);
         ctx->tx_key = ntohl(cmd->memory.key);
         ctx->tx_offset = 0;

